tests/e2e/test/pod-pvc-unexpected-reboot/chainsaw-test.yaml

```diff
@@ -0,0 +1,197 @@
+kind: Test
+metadata:
+  creationTimestamp: null
+  name: pod-pvc-unexpected-reboot
+  labels:
+    all:
+    basic:
+spec:
+  bindings:
+    - name: nodes
+      # number of nodes in cluster
+      value: (length(x_k8s_list($client, 'v1', 'Node', '').items))
+    - name: linode_url
+      value: (base64_decode(x_k8s_get($client, 'v1', 'Secret', 'kube-system', 'linode').data.apiurl || base64_encode('https://api.linode.com')))
+    - name: linode_token
+      value: (base64_decode(x_k8s_get($client, 'v1', 'Secret', 'kube-system', 'linode').data.token))
+  steps:
+    - name: Check if CSI Driver is deployed
+      try:
+        - assert:
+            file: assert-csi-driver-resources.yaml
+    - name: Create PVC and Pod
+      try:
+        - apply:
+            file: create-pvc-pod.yaml
+      catch:
+        - describe:
+            apiVersion: v1
+            kind: Pod
+        - describe:
+            apiVersion: v1
+            kind: PersistentVolumeClaim
+    - name: Check if Pod is ready and Volume is mounted
+      try:
+        - assert:
+            file: assert-pvc-pod.yaml
+      catch:
+        - describe:
+            apiVersion: v1
+            kind: PersistentVolumeClaim
+        - describe:
+            apiVersion: v1
+            kind: Pod
+    - name: Check if volume is create
+      try:
+        - script:
+            env:
+              - name: TARGET_API
+                value: ($linode_url)
+              - name: TARGET_API_VERSION
+                value: v4
+              - name: URI
+                value: volumes
+              - name: LINODE_TOKEN
+                value: ($linode_token)
+              - name: FILTER
+                value: (to_string({"tags":($namespace)}))
+            content: |
+              set -e
+              curl -s \
+                -H "Authorization: Bearer ${LINODE_TOKEN}" \
+                -H "X-Filter: $FILTER" \
+                -H "Content-Type: application/json" \
+                "${TARGET_API}/${TARGET_API_VERSION}/${URI}"
+            check:
+              ($error): ~
+              (json_parse($stdout)):
+                results: 1
+    - name: Create a file inside the pod and check it was created
+      try:
+        - script:
+            env:
+              - name: NAMESPACE
+                value: ($namespace)
+            content: |
+              kubectl exec -n $NAMESPACE e2e-pod -- sh -c "cd data && touch testfile" && \
+              kubectl exec -n $NAMESPACE e2e-pod -- sh -c "ls data"
+            check:
+              ($error): ~
+              (contains($stdout, 'testfile')): true
+
+    - name: Create STS
+      try:
+        - apply:
+            file: create-sts.yaml
+        - assert:
+            file: assert-sts.yaml
+        - script:
+            env:
+              - name: NAMESPACE
+                value: ($namespace)
+            content: |
+              kubectl exec -n $NAMESPACE redis-test-0 -- sh -c "cd data && touch testfile"
```

Comment:
what is purpose of the redis-test? The node is not rebooted.

Reply:
tests split and specific node rebooted :)